src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/ExportClientResponse.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+
+internal abstract class ExportClientResponse
+{
+    protected ExportClientResponse(bool success, DateTime deadline, Exception? exception)
+    {
+        this.Success = success;
+        this.Exception = exception;
+        this.Deadline = deadline;
+    }
+
+    public bool Success { get; }
+
+    public Exception? Exception { get; }
+
+    public DateTime? Deadline { get; }
```

Comment:
Seems like this should be `public DateTime Deadline { get; }` (not `DateTime?`)? Also might be good to rename it `DeadlineUtc` I feel like that makes things easier to read / less error prone 😄

Reply:
Apply the same to local variables in other methods too.